hr_expense_petty_cash/models/account_invoice.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    is_petty_cash = fields.Boolean(
+        string='Petty Cash',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.multi
+    @api.constrains('invoice_line_ids')
+    def _check_petty_cash_amount(self):
+        for rec in self:
+            if not rec.is_petty_cash:
+                continue
+            petty_cash = self.env['petty.cash'].search(
```

Comment:
i think you should move `self.env['petty.cash']` to top of function, it will improve the performance.